### PR TITLE
Feature/crd autoscaler status

### LINKED ISSUE #886: Bug: InputManifest Status while autoscaling
Frontend is not updating the `inputManifest` statuses when a build is triggered by the autoscaler. The reason for this is that the state change does not go through the frontend. This can be fixed by sending a [Generic Event](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/event#GenericEvent) to the controller from the `autoscaler adapter` with `inputmanifest` resource `name` and `namespace`.

### PR DESCRIPTION
**Pull request by cloudziu**
This PR will fix #886 and #912 

- add InputManifest reference fields in proto
- add frontend grpc server to listen for events from autoscaler-adapter
- add autoscaler-adapter frontend grpc client to send events

### CODE REVIEW

File: internal/envs/envs.go [hunk 1]
```diff
@@ -64,6 +66,11 @@ func init() {
 	}
 	ContextBoxURL = strings.ReplaceAll(ContextBoxURL, ":tcp://", "")
 
+	if FrontendURL == ":" {
+		FrontendURL = "localhost:50055"
```
Comment (MiroslavRepka) [inline internal/envs/envs.go:70]: I believe this is address of `context-box`

File: proto/frontend.proto [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+message SendAutoscalerEventRequest {
+    string InputManifestName = 1;
+    string InputManifestNamespace = 2;
+}
+
+message SendAutoscalerEventResponse {}
+
+service FrontendService {
+    // SendAutoscalerEvent sends events to frontend when a scale up/down encures
```
Comment (MiroslavRepka) [inline proto/frontend.proto:14]: ~`encures`~ -> `occurs`

File: services/autoscaler-adapter/claudie_provider/claudie_node_groups.go [hunk 3]
```diff
@@ -182,6 +182,24 @@ func (c *ClaudieCloudProvider) updateNodepool(nodepool *pb.NodePool) error {
 	return nil
 }
 
+func (c *ClaudieCloudProvider) SendAutoscalerEvent() error {
```
Comment (MiroslavRepka) [inline services/autoscaler-adapter/claudie_provider/claudie_node_groups.go:185]: I think it makes more sense to have it in `claudie_provider.go`

File: services/autoscaler-adapter/claudie_provider/claudie_provider.go [hunk 4]
```diff
@@ -41,6 +41,10 @@ type ClaudieCloudProvider struct {
 
 	// Name of the Claudie config.
 	projectName string
+	// Kubernete InputManifest resource name
```
Comment (MiroslavRepka) [inline services/autoscaler-adapter/claudie_provider/claudie_provider.go:44]: ~`Kubernete`~ -> `Kubernetes`

File: services/frontend/client/client_test.go [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+func closeConn(t *testing.T, connection *grpc.ClientConn) {
+	err := connection.Close()
+	if err != nil {
+		log.Fatal().Msgf("Error while closing the client connection: %v", err)
+	}
+	require.NoError(t, err)
+}
+
+func TestGetConfigBuilder(t *testing.T) {
```
Comment (MiroslavRepka) [inline services/frontend/client/client_test.go:33]: This should probably have different name

File: services/frontend/main.go [hunk 6]
```diff
@@ -57,19 +62,42 @@ func run() error {
 		return err
 	}
 
+	autoscalerChan := make(chan event.GenericEvent)
```
Comment (MiroslavRepka) [inline services/frontend/main.go:65]: Are we closing it anywhere? If not, maybe we should as a part of a graceful shutdown.

File: services/frontend/pkg/controller/controller.go [hunk 7]
```diff
@@ -252,14 +254,14 @@ func (r *InputManifestReconciler) Reconcile(ctx context.Context, req ctrl.Reques
...
-	if err := r.Usecases.CreateConfig(im); err != nil {
+func (r *InputManifestReconciler) createConfig(im *manifest.Manifest, resourceName string, resourceNamespace string) error {
+	if err := r.Usecases.CreateConfig(im, resourceName, resourceNamespace); err != nil {
 		return err
 	}
 	return nil
 }
 
-func (r *InputManifestReconciler) deleteConfig(im *manifest.Manifest) error {
+func (r *InputManifestReconciler) deleteConfig(im *manifest.Manifest, resourceName string, resourceNamespace string) error {
```
Comment (MiroslavRepka) [inline services/frontend/pkg/controller/controller.go:264]: Are the new parameters used anywhere?

File: services/frontend/server/adapters/inbound/grpc/adapter.go [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/health"
+
+	"github.com/berops/claudie/internal/utils"
+	"github.com/berops/claudie/proto/pb"
+	"github.com/berops/claudie/services/frontend/server/domain/usecases"
+)
+
+const (
+	defaultContextBoxPort = 50055
```
Comment (MiroslavRepka) [inline services/frontend/server/adapters/inbound/grpc/adapter.go:17]: I think this should be a frontend port.

File: services/frontend/server/adapters/inbound/grpc/adapter.go [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+
+type GrpcAdapter struct {
+	tcpListener       net.Listener
+	server            *grpc.Server
+	healthCheckServer *health.Server
+}
+
+// Init will create the underlying gRPC server and the gRPC healthcheck server
+func (g *GrpcAdapter) Init(usecases *usecases.Usecases) {
+	port := utils.GetEnvDefault("FRONTEND_PORT", fmt.Sprint(defaultContextBoxPort))
```
Comment (MiroslavRepka) [inline services/frontend/server/adapters/inbound/grpc/adapter.go:28]: ditto here.

File: services/frontend/main.go (hunk 6 above)
Comment (cloudziu) [reply services/frontend/main.go:65]: Added 👍

File: services/frontend/pkg/controller/controller.go [hunk 10]
```diff
@@ -22,6 +22,8 @@ func (r *InputManifestReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 	log := crlog.FromContext(ctx)
 	inputManifest := &v1beta.InputManifest{}
 
+	log.Info(req.NamespacedName.String())
```
Comment (MiroslavRepka) [inline services/frontend/pkg/controller/controller.go:25]: Are we keeping this log? If so, maybe some additional context is needed.

File: services/kuber/server/autoscaler/cluster_autoscaler.go [hunk 11]
```diff
@@ -93,6 +98,8 @@ func (ab *AutoscalerBuilder) generateFiles() error {
 	// Prepare data
 	clusterId := fmt.Sprintf("%s-%s", ab.cluster.ClusterInfo.Name, ab.cluster.ClusterInfo.Hash)
 	version, err := getK8sVersion(ab.cluster.Kubernetes)
+	frontendHostname := utils.GetEnvDefault("FRONTEND_HOSTNAME", fmt.Sprint(defaultFrontendHostname))
```
Comment (MiroslavRepka) [inline services/kuber/server/autoscaler/cluster_autoscaler.go:101]: the `fmt.Sprint` is redundant here (also on the line below)

**Review by MiroslavRepka — APPROVED** [review]
LGTM left just some minor comments. Good work @cloudziu!

**Pull request merged**